Extensions/Spine/SpineObjectConfiguration.h

```diff
@@ -0,0 +1,163 @@
+    return gd::make_unique<SpineObjectConfiguration>(*this);
+  }
+
+  virtual void ExposeResources(gd::ArbitraryResourceWorker &worker) override;
+  
+  virtual std::map<gd::String, gd::PropertyDescriptor>GetProperties() const override;
+  
+  virtual bool UpdateProperty(const gd::String &name, const gd::String &value) override;
+
+  virtual std::map<gd::String, gd::PropertyDescriptor>
+  GetInitialInstanceProperties(const gd::InitialInstance &instance,
+                               gd::Project &project,
+                               gd::Layout &layout) override;
+
+  virtual bool UpdateInitialInstanceProperty(gd::InitialInstance &instance,
+                                             const gd::String &name,
+                                             const gd::String &value,
+                                             gd::Project &project,
+                                             gd::Layout &layout) override;
+
+  /** \name Animations
+   * Methods related to animations management
+   */
+  ///@{
+  /**
+   * \brief Return the animation at the specified index.
+   * If the index is out of bound, a "bad animation" object is returned.
+   */
+  const SpineAnimation &GetAnimation(std::size_t nb) const;
+
+  /**
+   * \brief Return the animation at the specified index.
+   * If the index is out of bound, a "bad animation" object is returned.
+   */
+  SpineAnimation &GetAnimation(std::size_t nb);
+
+  /**
+   * \brief Return the number of animations this object has.
+   */
+  std::size_t GetAnimationsCount() const { return animations.size(); };
+
+  /**
+   * \brief Return true if the animation called "name" exists.
+   */
+  bool HasAnimationNamed(const gd::String& name) const;
+
+  /**
+   * \brief Add an animation at the end of the existing ones.
+   */
+  void AddAnimation(const SpineAnimation &animation);
+
+  /**
+   * \brief Remove an animation.
+   */
+  bool RemoveAnimation(std::size_t nb);
+
+  /**
+   * \brief Remove all animations.
+   */
+  void RemoveAllAnimations() { animations.clear(); }
+
+  /**
+   * \brief Return true if the object hasn't any animation.
+   */
+  bool HasNoAnimations() const { return animations.empty(); }
+
+  /**
+   * \brief Swap the position of two animations
+   */
+  void SwapAnimations(std::size_t firstIndex, std::size_t secondIndex);
+
+  /**
+   * \brief Change the position of the specified animation
+   */
+  void MoveAnimation(std::size_t oldIndex, std::size_t newIndex);
+
+  /**
+   * \brief Return a read-only reference to the vector containing all the
+   * animation of the object.
+   */
+  const std::vector<SpineAnimation> &GetAllAnimations() const {
+    return animations;
+  }
+
+  ///@}
+
+protected:
+  virtual void DoUnserializeFrom(gd::Project &project, const gd::SerializerElement &element) override;
+  virtual void DoSerializeTo(gd::SerializerElement &element) const override;
+
+private:
+  double scale;
+  double opacity;
```

Comment:
I guess the opacity property could be removed. Especially because the transparency doesn't render correctly. The object parts are blending into each other which make them less transparent.

Reply:
I went ahead and removed the opacity property too, but in its own commit so it's easy to create a revert commit.